Bump aioesphomeapi to 17.0.1

bdraco opened the pull request:
# What does this implement/fix?

fixes https://github.com/home-assistant/core/issues/101039
Fixes https://github.com/esphome/issues/issues/4936

An alternative fix would also be to bump zeroconf to 0.115.1

bug fixes for zeroconf resolution

changelog: https://github.com/esphome/aioesphomeapi/compare/v17.0.0...v17.0.1

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

**Related issue or feature (if applicable):** fixes <link to issue>

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#<esphome-docs PR number goes here>

## Test Environment

- [ ] ESP32
- [ ] ESP32 IDF
- [x] ESP8266
- [ ] RP2040

## Example entry for `config.yaml`:
<!--
  Supplying a configuration snippet, makes it easier for a maintainer to test
  your PR. Furthermore, for new integrations, it gives an impression of how
  the configuration would look like.
  Note: Remove this section if this PR does not have an example entry.
-->

```yaml
# Example config.yaml

```

## Checklist:
  - [ ] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

bdraco committed:
Bump aioesphomeapi to 17.0.1

fixes https://github.com/home-assistant/core/issues/101039

jesserockz commented:
The `cryptography` package is holding this back, but it also has a native apt package that is installed and I have not been able to test it successfully on the different architectures.

#5450

bdraco commented:
Should we bump zeroconf to 0.115.1 and backport instead to fix the linked issue?

jesserockz commented:
Yeah if that will fix it, I can push that bump out #5470

bdraco commented:
> Yeah if that will fix it, I can push that bump out #5470

Yes it will.  I fixed it the problem there and I also updated aioesphomeapi to use the newer api that does not have the problem so either bump will solve it

bdraco commented:
replaced by #5557
